springcloud-streams/src/main/java/org/axonframework/extensions/springcloud/eventhandling/SpringStreamMessageSource.java

```diff
@@ -0,0 +1,150 @@
+
+    /**
+     * Instantiate a {@link SpringStreamMessageSource} based on the fields contained in the {@link SpringStreamMessageSource.Builder}.
+     * The {@link EventBus} is a <b>hard requirement</b> and thus should be provided.
+     * The {@link SpringMessageConverter} is a <b>hard requirement</b> and thus should be provided.
+     * <p>
+     * Will validate that the {@link EventBus} and {@link SpringMessageConverter} are not {@code null}, and will throw an
+     * {@link AxonConfigurationException} if for either of them this holds.
+     *
+     * @param builder the {@link SpringStreamMessageSource.Builder} used to instantiate a {@link SpringStreamMessageSource} instance
+     */
+    protected SpringStreamMessageSource(SpringStreamMessageSource.Builder builder) {
+        builder.validate();
+        if (builder.eventBus != null) {
+            this.messageProcessors.addAll(singletonList(builder.eventBus::publish));
+        }
+        this.converter = builder.converter;
+    }
+
+    /**
+     * Instantiate a Builder to be able to create a {@link SpringStreamMessageSource}.
+     * The {@link SpringMessageConverter} is a <b>hard requirement</b> and thus should be provided.
+     *
+     * @return a Builder to be able to create a {@link SpringStreamMessageSource}.
+     */
+    public static SpringStreamMessageSource.Builder builder() {
+        return new SpringStreamMessageSource.Builder();
+    }
+
+    @Override
+    public Registration subscribe(Consumer<List<? extends EventMessage<?>>> messageProcessor) {
+        messageProcessors.add(messageProcessor);
+        return () -> messageProcessors.remove(messageProcessor);
+    }
+
+    /**
+     * Handles the given {@link Message}. If the filter refuses the message, it is ignored.
+     *
+     * @param message The spring message containing the event to publish
+     */
+    @Override
+    protected void handleMessageInternal(Message<?> message) {
+        Optional<EventMessage<?>> optional = converter.readSpringMessage(message);
+        optional.ifPresent(eventMessage -> {
+            List<? extends EventMessage<?>> messages = singletonList(eventMessage);
+            for (Consumer<List<? extends EventMessage<?>>> messageProcessor : messageProcessors) {
+                messageProcessor.accept(messages);
+            }
+        });
+    }
+
+    /**
+     * Builder class to instantiate a {@link SpringStreamMessageSource}.
+     * The {@link SpringMessageConverter} is a <b>hard requirement</b> and thus should be provided.
+     */
+    public static class Builder {
+
+        private EventBus eventBus;
+        private SpringMessageConverter converter;
+
+        /**
+         * Sets the eventBus for publishing events.
+         *
+         * @param eventBus The messageSource to subscribe
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public SpringStreamMessageSource.Builder eventBus(EventBus eventBus) {
```

Comment:
I think we can remove the `EventBus` from the builder, as described earlier in the constructor.